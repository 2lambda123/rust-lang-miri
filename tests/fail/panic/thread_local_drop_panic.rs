//@error-in-other-file: aborted execution

pub struct NoisyDrop {}

impl Drop for NoisyDrop {
    fn drop(&mut self) {
        panic!("ow");
    }
}

thread_local! {
    pub static NOISY: NoisyDrop = NoisyDrop {};
}

fn main() {
    NOISY.with(|_| ());
}
