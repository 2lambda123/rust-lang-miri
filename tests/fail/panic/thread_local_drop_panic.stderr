thread '<unnamed>' panicked at 'ow', $DIR/thread_local_drop_panic.rs:LL:CC
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: thread local panicked on drop
error: abnormal termination: the program aborted execution
  --> RUSTLIB/std/src/sys/PLATFORM/mod.rs:LL:CC
   |
LL |     unsafe { libc::abort() }
   |              ^^^^^^^^^^^^^ the program aborted execution
   |
   = note: inside `std::sys::PLATFORM::abort_internal` at RUSTLIB/std/src/sys/PLATFORM/mod.rs:LL:CC
   = note: inside `std::sys::PLATFORM::thread_local::fast_local::destroy_value::<NoisyDrop>` at RUSTLIB/std/src/rt.rs:LL:CC
   = note: inside `std::sys_common::thread_local_dtor::register_dtor_fallback::run_dtors` at RUSTLIB/std/src/sys_common/thread_local_dtor.rs:LL:CC

error: aborting due to previous error

